fixtures/tests/_fixtures/test_warnings.py

```diff
@@ -47,3 +47,28 @@ def test_capture_category(self):
         for i, category in enumerate(categories):
             c = w.captures[i]
             self.assertEqual(category, c.category)
+
+
+class TestWarningsFilter(testtools.TestCase, fixtures.TestWithFixtures):
+
+    def test_filter(self):
+        fixture = fixtures.WarningFilter(
+            [
+                {
+                    'action': 'ignore',
+                    'category': 'DeprecationWarning',
+                },
+                {
+                    'action': 'once',
+                    'category': 'UserWarning',
+                },
+            ],
+        )
+        self.useFixture(fixture)
+        with warnings.catch_warnings(record=True) as w:
+            warnings.warn('deprecated', DeprecationWarning)
+            warnings.warn('user', UserWarning)
+
+        # only the user warning should be present, and it should only have been
+        # raised once
+        self.assertEqual(1, len(w))
```

Comment:
Could we have a quick test here that the original filters have been properly restored?  You said that doing this was the main point of having this fixture here, and I agree that's a good reason, so it seems that it would be worthwhile to explicitly test that behaviour.

Reply:
Done.